### PR TITLE
feat: Add APIs for linear integration

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-3279/add-get-apis

- [x] Added Get Team API.
- [x] Added Get Team entities API.
- [x] Create linear issue.
- [x] Link issue (Attachment conversation URL with issue id)
- [x] Unlink issue.
- [x] Get all the linked issues from the conversation url.
- [x] Search linear issues.

### CODE REVIEW

File: lib/linear_queries.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          name
+        }
+      }
+    }
+  GRAPHQL
+
+  def self.team_entites_query(team_id)
+    <<~GRAPHQL
+      query {
+        users {
```
Comment (muhsin-k) [inline lib/linear_queries.rb:16]: The user can be associated to multiple teams.

File: lib/linear_queries.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          nodes {
+            id
+            name
+          }
+        }
+        projects {
```
Comment (muhsin-k) [inline lib/linear_queries.rb:22]: The projects can be associated to multiple teams.

**Comment by sojan-official** [comment]
The code, in general, looks good to me. For the following steps, 
- [ ] Let's finalise the flow and test out the changes over this PR. 
- [ ] Extract the `Linear` class into a separate gem
- [ ] Break down the changes into smaller PRs and get this merged.

**Review by sojan-official — APPROVED** [review]
LGTM

- verified in local over the feature branch

**Pull request merged**